exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/validate/IteratorValidatorBatchIterator.java

```diff
@@ -257,7 +264,9 @@ public IterOutcome next() {
           // NONE is allowed even without seeing a OK_NEW_SCHEMA. Such NONE is called
           // FAST NONE.
           // NONE moves to terminal high-level state.
-          validationState = ValidationState.TERMINAL;
+          if (!isRepeatable) {
```

Comment:
Pls modify the comment's last line to indicate moving to TERMINAL state is conditional.

Reply:
Done